sdk/src/main/scala/com/horizen/account/mempool/TxCache.scala

```diff
@@ -69,14 +94,50 @@ class TxCache {
 
   def getSizeInSlots: Int = sizeInSlots
 
+  def getNonExecSizeInSlots: Int = nonExecSizeInSlots
+
+  // Returns the transaction that has been in the mempool for the longest time
   def getOldestTransaction(): Option[SidechainTypes#SCAT] = oldestTx.map(_.tx)
 
+  // Returns the latest transaction added to the mempool
   def getYoungestTransaction(): Option[SidechainTypes#SCAT] = youngestTx.map(_.tx)
 
+  def getNonExecIterator(): NonExecTransactionIterator = new NonExecTransactionIterator
 
-  case class TxMetaInfo(tx: SidechainTypes#SCAT) {
+  case class TxMetaInfo(tx: SidechainTypes#SCAT, var isNotExecutable: Boolean) {
     var next: Option[TxMetaInfo] = None
     var previous: Option[TxMetaInfo] = None
   }
 
+  class NonExecTransactionIterator {
+
+    private var nextElem: Option[TxMetaInfo] = oldestTx
+
+    @tailrec
+    final private[mempool] def findNext(txInfo: Option[TxMetaInfo]): Option[TxMetaInfo] = {
+      if (txInfo.isEmpty || txInfo.get.isNotExecutable)
+        txInfo
+      else {
+        findNext(txInfo.get.next)
+      }
+    }
+
+    def hasNext: Boolean = {
```

Comment:
I see that we just use it in UT, that is because we are using this iterator only when we are sure (comparing slot sizes) there is at least a non exec tx in mempool, correct?

Reply:
Correct. In fact, this method could be removed. I have left it just because iterators have hasNext.